### PR TITLE
PXP-628: [CLI]: Collect debug logs to a file

### PR DESCRIPTION
**Pull request by mfauzaan**
## Summary
Implement a feature to store logs when user enters the command with `--report` flag. Which we can use for debugging purposes.
Logs is stored in `~/.config/wukong/debug_log`

Note: Current logic only stores the debugging logs send from `wukong` 

Ticket: https://mindvalley.atlassian.net/browse/PXP-628

## What's Changed

- [x] Add a new flag `--report` for collecting logs
- [x] Add logic to store debug logs to a file

### CODE REVIEW

**Comment by sauron-droid** [comment]
[APPROVALNOTIFIER] This PR is **NOT APPROVED**

This pull-request has been approved by: *<a href="https://github.com/mindvalley/wukong-cli/pull/163#" title="Author self-approved">mfauzaan</a>*


The full list of commands accepted by this bot can be found [here](https://go.k8s.io/bot-commands?repo=mindvalley%2Fwukong-cli).

<details open>
Needs approval from an approver in each of these files:

- **[OWNERS](https://github.com/mindvalley/wukong-cli/blob/main/OWNERS)**

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

File: cli/src/auth/okta.rs
Comment (onimsha) [inline cli/src/auth/okta.rs:265]: I think we should also dump the whole error object into debug log. What do you think ?

File: cli/src/auth/okta.rs
Comment (onimsha) [inline cli/src/auth/okta.rs:352]: If the refresh token is expired, we can't get the following fields
* username
* exp
Only the active field is available, so we will need to handle it otherwise it will be a panic in the client side. I think you should add a test to check this behaviour.

Comment (onimsha) [reply cli/src/auth/okta.rs:352]: This is example from Okta's documentation 

image

Comment (mfauzaan) [reply cli/src/auth/okta.rs:353]: This handling isn't require because these values are optional fields in the response other then active.

Tested with invalid and here is the result:
`introspect response: refresh_token: asdfsadfasa, email: None, active: false, exp: None`

**Review by onimsha — APPROVED** [review]

**Comment by sauron-droid** [comment]
[APPROVALNOTIFIER] This PR is **NOT APPROVED**

This pull-request has been approved by: *<a href="https://github.com/mindvalley/wukong-cli/pull/163#" title="Author self-approved">mfauzaan</a>*, *<a href="https://github.com/mindvalley/wukong-cli/pull/163#pullrequestreview-1711771488" title="Approved">onimsha</a>*


The full list of commands accepted by this bot can be found [here](https://go.k8s.io/bot-commands?repo=mindvalley%2Fwukong-cli).

<details open>
Needs approval from an approver in each of these files:

- **[OWNERS](https://github.com/mindvalley/wukong-cli/blob/main/OWNERS)**

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

**Pull request merged**
